benchmarks/imagenet/resnet50/mocov2.py

```diff
@@ -0,0 +1,145 @@
+import copy
+from typing import List, Tuple
+
+import torch
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.nn import Identity
+from torch.optim import SGD
+from torchvision.models import resnet50
+
+from lightly.loss import NTXentLoss
+from lightly.models.modules import MoCoProjectionHead
+from lightly.models.utils import (
+    batch_shuffle,
+    batch_unshuffle,
+    get_weight_decay_parameters,
+    update_momentum,
+)
+from lightly.transforms import MoCoV2Transform
+from lightly.utils.benchmarking import OnlineLinearClassifier
+from lightly.utils.scheduler import CosineWarmupScheduler
+
+
+class MoCoV2(LightningModule):
+    def __init__(self, batch_size_per_device: int, num_classes: int) -> None:
+        super().__init__()
+        self.save_hyperparameters()
+        self.batch_size_per_device = batch_size_per_device
+
+        resnet = resnet50()
+        resnet.fc = Identity()  # Ignore classification head
+        self.backbone = resnet
+        self.projection_head = MoCoProjectionHead()
+        self.query_backbone = copy.deepcopy(self.backbone)
+        self.query_projection_head = MoCoProjectionHead()
+        self.criterion = NTXentLoss(
```

Comment:
Looking at the NTXentLoss function, it seems that the memory bank size should be an int, not a tuple

Reply:
This PR also included changes to the memory bank to make it accept a (num_features, feature_dim) tuple instead of a single `num_features` argument. The reason is that we have to know the `feature_dim` when creating the memory bank for distributed processing. Before, we would dynamically get the `feature_dim` from the first batch stored in the bank but in a distributed setting you cannot dynamically change the parameter size.